internal/controllers/port/actuator.go

```diff
@@ -157,6 +158,9 @@ func (actuator portCreateActuator) CreateResource(ctx context.Context, obj *orcv
 	// We explicitly disable default security groups by passing an empty
 	// value whenever the user does not specifies security groups
 	securityGroups := make([]string, len(resource.SecurityGroupRefs))
+	if len(securityGroups) > 0 && resource.PortSecurityEnabled != nil && !*resource.PortSecurityEnabled {
+		return nil, nil, orcerrors.Terminal(orcv1alpha1.ConditionReasonInvalidConfiguration, "SecurityGroupRefs cannot be set when PortSecurityEnabled is false")
```

Comment:
As mentioned in the API we should add API validation in this PR to reject an invalid configuration when PortSecurity is set explicitly.

When it isn't set explicitly, returning a terminal error is fine in principal. It occurs to me that we can (and probably do already) wait for the Network to be ready before executing this code. Is the inherited value of PortSecurityEnabled readable from the network status? If so we could take it into account in this validation.

Reply:
I tried to build a CEL today and failed to find the right syntax because it's a boolean. I'll try again next week.